src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.addAlive(this);
+  }
+
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeDead() {
+    return Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
It's better to change the hidden value to oposite one but not necessary

```suggestion
    this.hidden = !this.hidden;
```